File: Python/Binary Search Tree/L_BSTToGST.py
class Solution(object):
    def __init__(self):
        self.sum = 0
    def bstToGst(self, root):
        if root:
            self.bstToGst(root.right)  # Traverse the right subtree
            self.sum += root.val  # Update the sum
            root.val = self.sum  # Update the current node's value
            self.bstToGst(root.left)  # Traverse the left subtree
        return root

File: JavaScript/BinaryTree/L_BSTToGST.js
var bstToGst = function (root) {
    let sum = 0;

    const traverse = (node) => {
        if (node) {
            traverse(node.right); // Traverse the right subtree
            sum += node.val; // Update the sum
            node.val = sum; // Update the current node's value
            traverse(node.left); // Traverse the left subtree
        }
    };

    traverse(root);
    return root;
};
